core/primitives/src/errors.rs

```diff
@@ -21,6 +21,8 @@ pub enum TxExecutionError {
     InvalidTxError(InvalidTxError),
 }
 
+impl std::error::Error for TxExecutionError {}
```

Comment:
To be clear, `TxExecutionError` isn't directly an `Rpc*Error` but is instead embedded deep inside `FinalExecutionOutcomeView` that's the `Rpc*Response` of both the `broadcast_tx_commit` and `tx` RPC methods.

Reply:
Hm, taking a second look here, this impl in particular feels suspect to me. `TxExecutionError` should **not** be an `std::error::Error`, it should **not** work with the `?` syntax. 

It's not a runtime error a-la io error. It is a value which signifies that the tx was succesfully processed, and that, as a result of this processing, the runtime concluded that the tx is invalid.